### PR TITLE
Check if /dev/log exists before initializing syslog logging (fix #282)

### LINKED ISSUE #282: logging setup should explicitly check if `/dev/log` exists
Currently on macOS the check at https://github.com/PyAr/fades/blob/master/fades/logger.py#L54 passes, since the handler initialization does not check if `/dev/log` exists. Later when you actually run, the console is spammed heavily with errors:

```
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/logging/handlers.py", line 857, in _connect_unixsocket
    self.socket.connect(address)
FileNotFoundError: [Errno 2] No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/logging/handlers.py", line 939, in emit
    self._connect_unixsocket(self.address)
  File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/logging/handlers.py", line 868, in _connect_unixsocket
    self.socket.connect(address)
FileNotFoundError: [Errno 2] No such file or directory
Call stack:
  File "/usr/local/Cellar/fades/6.0.1_2/libexec/bin/fades", line 33, in <module>
    main.go(sys.argv)
  File "/usr/local/Cellar/fades/6.0.1_2/libexec/lib/python3.6/site-packages/fades/main.py", line 257, in go
    options, pip_options)
  File "/usr/local/Cellar/fades/6.0.1_2/libexec/lib/python3.6/site-packages/fades/envbuilder.py", line 148, in create_venv
    destroy_venv(env_path)
  File "/usr/local/Cellar/fades/6.0.1_2/libexec/lib/python3.6/site-packages/fades/envbuilder.py", line 171, in destroy_venv
    logger.debug("Destroying virtualenv at: %s", env_path)
Message: 'Destroying virtualenv at: %s'
```
A possible solution IMO is to just check existence of `/dev/log` and avoid pushing the syslog handler if it does not exist.

### PR DESCRIPTION
**Pull request by vmalloc**

### CODE REVIEW

**Comment by facundobatista** [comment]
Hello! Thanks for this PR!!

File: fades/logger.py [hunk 1]
```diff
@@ -50,15 +50,17 @@ def set_up(verbose, quiet):
...
-    except:
-        # silently ignore that the user doesn't have a syslog active; can
-        # see all the info with "-v" anyway
-        pass
-    else:
-        logger.addHandler(handler)
-        formatter = logging.Formatter(FMT_SYSLOG)
-        handler.setFormatter(formatter)
+    syslog_path = '/dev/log'
+    if os.path.exists(syslog_path):
```
Comment (facundobatista) [inline fades/logger.py]: You'd need to import os.path in the top of the file for this to work

**Comment by facundobatista** [comment]
Note that travis checks failed, there's a missing import (I annotated that in a review).

You can easily check this locally, just run `./test`

**Comment by vmalloc** [comment]
Oops, that's embarrassing... Fixing it and pushing again in a moment...

**Comment by vmalloc** [comment]
Fixed, tested and pushed again. Sorry for that...

**Comment by facundobatista** [comment]
Awesome, thanks for this work!

**Comment by facundobatista** [comment]
Awesome, thanks for this work! We're having problems to merge this, because Travis is failing on nightly on a totally (it looks) unrelated problem, but we'll fix it ASAP and get your code in.

Thanks again!

**Comment by vmalloc** [comment]
Thanks! Will it be possible to release a fix with this change afterwards? Using `fades` from `brew` is really spammy and annoying at the moment on a Mac...

**Comment by gilgamezh** [comment]
@vmalloc please update this PR with master to have Travis working

**Comment by vmalloc** [comment]
@gilgamezh done - rebased and pushed again

**Comment by vmalloc** [comment]
@facundobatista I resolved all pending review comments and updated based on the issue discussion. FYI

File: fades/logger.py [hunk 2]
```diff
@@ -50,15 +51,20 @@ def set_up(verbose, quiet):
...
-        # see all the info with "-v" anyway
-        pass
-    else:
-        logger.addHandler(handler)
-        formatter = logging.Formatter(FMT_SYSLOG)
-        handler.setFormatter(formatter)
+    for syslog_path in ('/dev/log', '/var/run/syslog'):
+        if not os.path.exists(syslog_path):
+            continue
+        if os.path.exists(syslog_path):
```
Comment (facundobatista) [inline fades/logger.py]: you should remove this `if` (as it's validated just above) and dedent the rest of the try/except...

Thanks!

Comment (vmalloc) [reply fades/logger.py]: @facundobatista whoops - a copy-paste error. Fixed and pushed again. Sorry!

**Comment by facundobatista** [comment]
Awesome, thank you very much!

**Pull request merged**